pkg/retriever/graphsyncretriever.go

```diff
@@ -405,12 +416,29 @@ func retrievalPhase(
 	timeout time.Duration,
 	candidate types.RetrievalCandidate,
 	queryResponse *retrievalmarket.QueryResponse,
+	selector ipld.Node,
 	eventsCallback datatransfer.Subscriber,
 ) (*types.RetrievalStats, error) {
+
+	if selector == nil {
```

Comment:
it seems odd to have this code in each retriever. What if someone makes a new retriever and forgets? I'm kind of inclined to move it up somehow?